Patches for atom 1.18

cdavidjean opened the pull request:
Fixed startup for atom 1.18
Fixed pub calls
Fixed linter/squiggles

devoncarew commented on lib/impl/tooltip.dart:
```diff
@@ -116,19 +131,24 @@ class TooltipElement extends CoreElement {
       : super('div', classes: 'hover-tooltip') {
     id = 'hover-tooltip';
 
+    print(position);
```
We should remove this.

cdavidjean replied:
Done.

devoncarew commented on package.json:
```diff
@@ -26,9 +26,9 @@
       }
     },
-    "linter-plus-self": {
+    "linter-indie": {
```
Do we need to update the required-packages list above as well?

Arcanemagus replied:
Depends on whether whatever is consuming the `required-packages` key supports version numbers as that service is only available with `linter` v2.

devoncarew commented:
I landed the atom.dart plugin, so the travis build job now finishes cleanly.

cdavidjean commented:
Oh damn...  like I said I'm new to git, i thought the pull request would
use a snapshot copy...  i am not done with reviving the tootips yet.
Should i try to uncommit those changes?

On Thu, Jul 27, 2017 at 5:03 PM Devon Carew <notifications@github.com>
wrote:

> *@devoncarew* commented on this pull request.
>
> Thanks for the PR! Much appreciated.
>
> I see you added some additional commits - is this PR still in progress?
> ------------------------------
>
> In lib/impl/tooltip.dart
> <https://github.com/dart-atom/dartlang/pull/1130#discussion_r129866943>:
>
> > @@ -116,19 +131,24 @@ class TooltipElement extends CoreElement {
>        : super('div', classes: 'hover-tooltip') {
>      id = 'hover-tooltip';
>
> +    print(position);
>
> We should remove this.
> ------------------------------
>
> In package.json
> <https://github.com/dart-atom/dartlang/pull/1130#discussion_r129867253>:
>
> > @@ -26,9 +26,9 @@
>          "^1.0.0": "consumeToolbar"
>        }
>      },
> -    "linter-plus-self": {
> +    "linter-indie": {
>
> Do we need to update the required-packages list above as well?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/dart-atom/dartlang/pull/1130#pullrequestreview-52684748>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AdDFw6iSkR7Qw_GKJq4J4qhnmy818m0Rks5sSKasgaJpZM4Ojrrg>
> .
>

devoncarew commented:
> Should i try to uncommit those changes?

I would either revert those changes and update the PR, or keeping working and I can re-review when you're done. You'll have a bit of a git dance to do, so the 2nd option would likely be easier for you (given that you're new to git. Happy either way :)

cdavidjean commented:
Thanks, I'll take option two :). It shouldn't be too long, probably some
time tomorrow.


On Thu, Jul 27, 2017 at 6:51 PM Devon Carew <notifications@github.com>
wrote:

> Should i try to uncommit those changes?
>
> I would either revert those changes and update the PR, or keeping working
> and I can re-review when you're done. You'll have a bit of a git dance to
> do, so the 2nd option would likely be easier for you (given that you're new
> to git. Happy either way :)
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/dart-atom/dartlang/pull/1130#issuecomment-318421181>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AdDFw5ODkepesE2Nn-JPCqby3uYlvTsFks5sSMAZgaJpZM4Ojrrg>
> .
>

cdavidjean commented:
A Better summary of this PR (which is now ready for review, sorry):
- Fixed jumpKey navigation
- Fixed linter service (using newer linter api)
- Fixed outline by attaching to editor element instead of shadowDom root.
- Fixed some startup NPE.
- Revived Tooltips, moved them above (not to clash with linter) and added debouncing.
- Added tracking subscription for TooltipManager.

devoncarew commented:
Awesome! Thanks much, landing.
